File: 1709 Amalgamated Artichokes.cpp
#include <cstdio>
#include <cmath>
#include <algorithm>

// #include <iostream>

using namespace std;

int main() {
    int p, a, b, c, d, n;
    while(scanf("%d %d %d %d %d %d", &p, &a, &b, &c, &d, &n) == 6) {
        double minValue = -1e+100, maxValue = 0, calc;
        for (int k = 1; k <= n; k++) {
            calc = p * (sin((a * k + b)) + cos((c * k + d)) + 2);
            // cout << "calc" << calc << endl;
            minValue = max(minValue, maxValue - calc);
            // cout << "minValue" << minValue << endl;
            maxValue = max(maxValue, calc);
            // cout << "maxValue" << maxValue << endl;
        }
        printf("%.6f\n", minValue);
    }
    return 0;
}

File: 1709 Amalgamated Artichokes.py
import math
# price(k) = p(sin(ak + b) + cos(ck + d) + 2)

while True:
    try:
        newInput = input()
        numeros = newInput.split()

        if len(numeros) >= 6:
            p = int(numeros[0])
            a = int(numeros[1])
            b = int(numeros[2])
            c = int(numeros[3])
            d = int(numeros[4])
            n = int(numeros[5])

        maxValue = -1e+100
        minValue = 0
        for k in range(n):
            calc = p*(math.sin(a*(k+1)+b)+math.cos(c*(k+1)+d)+2)
            minValue = max(minValue, maxValue - calc)
            maxValue = max(maxValue, calc)
        lastCalc = minValue
        fCalc = '{:.6f}'.format(lastCalc)
        print(fCalc)
    except EOFError:
        break